aips/aip-x.md

```diff
@@ -0,0 +1,707 @@
+
+### Message Construction
+
+The proposed plaintext representation will be an adaptation of the standards specified from [EIP-4361](https://eips.ethereum.org/EIPS/eip-4361) and [CAIP-122](https://chainagnostic.org/CAIPs/caip-122). If fields are not present, omit them from the plaintext message.
+
+#### Message Format
+
+```
+${domain} wants you to sign in with your Aptos account:
+${address}
+
+${statement}
+
+URI: ${uri}
+Version: ${version}
+Chain ID: ${chain-id}
+Nonce: ${nonce}
+Issued At: ${issued-at}
+Expiration Time: ${expiration-time}
+Not Before: ${not-before}
+Request ID: ${request-id}
+Resources:
+- ${resources[0]}
+- ${resources[1]}
+...
+- ${resources[n]}
+```
+
+#### Example Messages
+
+_Minimal Example_
+
+```
+example.com wants you to sign in with your Aptos account:
+0x0000000000000000000000000000000000000000000000000000000000000001
+
+URI: https://example.com
+Version: 1
+Chain ID: mainnet
+```
+
+_Detailed Example_
+
+```
+example.com wants you to sign in with your Aptos account:
+0x0000000000000000000000000000000000000000000000000000000000000001
+
+Sign in to Example.com
+
+URI: https://example.com
+Version: 1
+Nonce: abc123
+Issued At: 2023-01-01T00:00:00Z
+Expiration Time: 2024-01-01T00:00:00Z
+Not Before: 2023-01-01T00:00:00Z
+Request ID: req123
+Chain ID: mainnet
+Resources:
+- resource1
+- resource2
+```
+
+### Augment Backus-Naur Form (ABNF) Expression
+
+A constructed SIWA message should conform the the following Augment Backus–Naur Form expression:
+
+```
+sign-in-with-aptos =
+message-domain %s" wants you to sign in with your Aptos account:" LF
+message-address
+[ LF LF message-statement ]
+[ LF advanced-fields ]
+
+advanced-fields =
+[ LF %s"URI: " message-uri ]
+[ LF %s"Version: " message-version ]
+[ LF %s"Chain ID: " message-chain-id ]
+[ LF %s"Nonce: " message-nonce ]
+[ LF %s"Issued At: " message-issued-at ]
+[ LF %s"Expiration Time: " message-expiration-time ]
+[ LF %s"Not Before: " message-not-before ]
+[ LF %s"Request ID: " message-request-id ]
+[ LF %s"Resources:" message-resources ]
+
+message-domain = authority
+message-address = "0x" 64*64HEXDIG
+message-statement = 1*( reserved / unreserved / " " )
+message-uri = URI
+message-version = "1"
+message-chain-id = %s"mainnet" / %s"testnet" / %s"devnet" / %s"localnet" / %s"aptos:mainnet" / %s"aptos:testnet" / %s"aptos:devnet" /  %s"aptos:" 1*DIGIT
+message-nonce = 8*( ALPHA / DIGIT )
+message-issued-at = date-time
+message-expiration-time = date-time
+message-not-before = date-time
+message-request-id = *pchar
+message-resources = \\*( LF "- " URI )
+```
+
+### Domain Separating Signing Algorithm
+
+A plaintext construction of a SIWA message must be prefixed with a valid SIWA hash before signing to ensure domain separation across other signatures. The messages must conform to the following expression:
+
+```
+sha3_256(b"SIGN_IN_WITH_APTOS::") || <siwa_message>
+```
+
```

Comment:
Maybe add the following clarification:

 > where `<siwa_message>` is a sign-in with Aptos message formatted as explained in ["Message Format"](#message-format) above.